lib/spack/docs/basic_usage_tutorial.rst

```diff
@@ -0,0 +1,1243 @@
+  | | | | | |\
+  | | | | | | |\
+  | o | | | | | |  netcdf
+  | |\ \ \ \ \ \ \
+  | | |\ \ \ \ \ \ \
+  | | | o | | | | | |  curl
+  | | |/| | | | | | |
+  | |/| | | | | | | |
+  | | | o | | | | | |  openssl
+  | | |/ / / / / / /
+  | |/| | | | | | |
+  | | o | | | | | |  hdf5
+  | |/| | | | | | |
+  | | |/ / / / / /
+  | o | | | | | |  zlib
+  |  / / / / / /
+  o | | | | | |  swig
+  o | | | | | |  pcre
+   / / / / / /
+  o | | | | |  mpi
+   / / / / /
+  o | | | |  matio
+   / / / /
+  o | | |  lapack
+   / / /
+  o | |  glm
+   / /
+  o |  boost
+   /
+  o  blas
+
+
+You can control how the output is displayed with a number of options.
+
+The ASCII output from ``spack graph`` can be difficult to parse for
+complicated packages. The output can be changed to the ``graphviz``
+``.dot`` format using the `--dot` flag.
+
+.. code-block:: console
+
+  $ spack graph --dot trilinos | dot -Tpdf trilinos_graph.pdf
+
+.. _basics-tutorial-uninstall:
+
+---------------------
+Uninstalling Packages
+---------------------
+
+Earlier we installed many configurations each of libelf and
+libdwarf. Now we will go through and uninstall some of those packages
+that we didn't really need.
+
+.. code-block:: console
+
+  $ spack find -d libdwarf
+  ==> 3 installed packages.
+  -- linux-redhat6-x86_64 / gcc@4.4.7 -----------------------------
+      libdwarf@20160507
+          ^libelf@0.8.12
+
+      libdwarf@20160507
+          ^libelf@0.8.13
+
+
+  -- linux-redhat6-x86_64 / intel@16.0.3 --------------------------
+      libdwarf@20160507
+          ^libelf@0.8.12
+
+  $ spack find libelf
+  ==> 6 installed packages.
+  -- linux-redhat6-x86_64 / gcc@4.4.7 -----------------------------
+  libelf@0.8.12
+  libelf@0.8.12
+  libelf@0.8.13
+
+  -- linux-redhat6-x86_64 / intel@15.0.4 --------------------------
+  libelf@0.8.13
+
+  -- linux-redhat6-x86_64 / intel@16.0.3 --------------------------
+  libelf@0.8.12
+  libelf@0.8.13
+
+
+We can uninstall packages by spec using the same syntax as install.
+
+.. code-block:: console
+
+  $ spack uninstall libelf%intel@15.0.4
+  ==> The following packages will be uninstalled :
+
+  -- linux-redhat6-x86_64 / intel@15.0.4 --------------------------
+  w33hrej libelf@0.8.13%intel
+
+
+  ==> Do you want to proceed ? [y/n]
+  y
+  ==> Successfully uninstalled libelf@0.8.13%intel@15.0.4 arch=linux-redhat6-x86_64-w33hrej
+
+
+
+  $ spack find -lf libelf
+  ==> 5 installed packages.
+  -- linux-redhat6-x86_64 / gcc@4.4.7 -----------------------------
+  ipggckv libelf@0.8.12%gcc
+
+  vrv2ttb libelf@0.8.12%gcc cppflags="-O3"
+
+  csrt4qx libelf@0.8.13%gcc
+
+
+  -- linux-redhat6-x86_64 / intel@16.0.3 --------------------------
+  4blbe3q libelf@0.8.12%intel
+
+  7wgp32x libelf@0.8.13%intel
+
+
+We can uninstall packages by referring only to their hash.
+
+
+We can use either ``-f`` (force) or ``-d`` (remove dependents as well) to
+remove packages that are required by another installed package.
+
+
+  $ spack uninstall /4blb
```

Comment:
This code block and the one below it are missing:
```yaml
.. code-block::
```
As a result, they aren't getting formatted properly in the documentation.